File: frontend/src/types/studentApp.ts
import type { FourDimensionScores, GapItem } from './matching';
import type { StudentProfileResponse, StudentResponse } from './student';

export interface StudentSessionRequest {
  email?: string;
  phone?: string;
  name?: string;
}

export interface StudentSessionResponse {
  student: StudentResponse;
  created: boolean;
  has_profile: boolean;
}

export interface JobSnapshot {
  title?: string;
  city?: string;
  company_name?: string;
  company_stage?: string;
  industries?: string[];
  benefits?: string[];
}

export interface StudentRecommendationItem {
  id: string;
  job_profile_id: string;
  role_id?: string;
  role_name?: string;
  role_category?: string;
  total_score: number;
  scores: FourDimensionScores;
  gaps: GapItem[];
  match_reasons: string[];
  job_snapshot?: JobSnapshot;
  created_at: string;
  updated_at: string;
}

export interface StudentRecommendationResponse {
  student_id: string;
  results: StudentRecommendationItem[];
}

export interface CareerPathResponse {
  student_id: string;
  job_profile_id: string;
  role_id?: string;
  role_name?: string;
  role_category?: string;
  path: Record<string, unknown>;
}

export interface ProfileCompletionQuestion {
  question_id: string;
  title: string;
  prompt: string;
  placeholder?: string;
  options: string[];
}

export interface ProfileCompletionSessionResponse {
  student_id: string;
  questions: ProfileCompletionQuestion[];
}

export interface ProfileCompletionAnswer {
  question_id: string;
  answer: string;
}

export interface ProfileCompletionApplyRequest {
  answers: ProfileCompletionAnswer[];
}

export interface ProfileCompletionApplyResponse {
  profile: StudentProfileResponse;
  applied_updates: string[];
}

File: backend/app/schemas/student_app.py
"""Schemas for the student-facing mobile application."""
from datetime import datetime
from typing import Any
from uuid import UUID

from pydantic import BaseModel, Field

from app.schemas.matching import FourDimensionScores, GapItem
from app.schemas.student import StudentProfileResponse, StudentResponse


class StudentSessionRequest(BaseModel):
    """Lightweight student session bootstrap request."""

    email: str | None = None
    phone: str | None = None
    name: str | None = None


class StudentSessionResponse(BaseModel):
    """Student session bootstrap response."""

    student: StudentResponse
    created: bool = False
    has_profile: bool = False


class JobSnapshot(BaseModel):
    """Lightweight job snapshot used by the mobile recommendation cards."""

    title: str | None = None
    city: str | None = None
    company_name: str | None = None
    company_stage: str | None = None
    industries: list[str] = Field(default_factory=list)
    benefits: list[str] = Field(default_factory=list)


class StudentRecommendationItem(BaseModel):
    """Aggregated recommendation item for the student mobile app."""

    id: UUID
    job_profile_id: UUID
    role_id: UUID | None = None
    role_name: str | None = None
    role_category: str | None = None
    total_score: float
    scores: FourDimensionScores
    gaps: list[GapItem] = Field(default_factory=list)
    match_reasons: list[str] = Field(default_factory=list)
    job_snapshot: JobSnapshot | None = None
    created_at: datetime
    updated_at: datetime


class StudentRecommendationResponse(BaseModel):
    """Recommendation list for the student mobile app."""

    student_id: UUID
    results: list[StudentRecommendationItem] = Field(default_factory=list)


class CareerPathResponse(BaseModel):
    """Student-facing career path payload."""

    student_id: UUID
    job_profile_id: UUID
    role_id: UUID | None = None
    role_name: str | None = None
    role_category: str | None = None
    path: dict[str, Any]


class ProfileCompletionQuestion(BaseModel):
    """One structured question for profile completion."""

    question_id: str
    title: str
    prompt: str
    placeholder: str | None = None
    options: list[str] = Field(default_factory=list)


class ProfileCompletionSessionResponse(BaseModel):
    """Question session for profile completion."""

    student_id: UUID
    questions: list[ProfileCompletionQuestion] = Field(default_factory=list)


class ProfileCompletionAnswer(BaseModel):
    """Single completion answer."""

    question_id: str
    answer: str


class ProfileCompletionApplyRequest(BaseModel):
    """Apply completion answers to a student profile."""

    answers: list[ProfileCompletionAnswer] = Field(default_factory=list)


class ProfileCompletionApplyResponse(BaseModel):
    """Updated profile after applying completion answers."""

    profile: StudentProfileResponse
    applied_updates: list[str] = Field(default_factory=list)
